src/fontra/client/core/fontra-menus.js

```diff
@@ -0,0 +1,222 @@
+import * as html from "./html-utils.js";
+import { translate } from "/core/localization.js";
+import { MenuBar } from "/web-components/menu-bar.js";
+
+const mapMenuItemKeyToFunction = {
+  File: getFileMenuItems,
+  Font: getFontMenuItems,
+  // "Edit": getEditMenuItems,
+  // "View": getViewMenuItems,
+  // "Glyph": getGlyphMenuItems,
+};
+
+export function makeFontraMenuBar(menuItemKeys, delegate) {
+  const menuBarArray = [getFontraMenuItems()]; // Fontra-Menu at the beginning.
+
+  for (const itemKey of menuItemKeys) {
+    const methodName = `get${itemKey}MenuItems`;
+    if (typeof delegate[methodName] === "function") {
+      menuBarArray.push(delegate[methodName]());
+    } else if (mapMenuItemKeyToFunction[itemKey]) {
+      menuBarArray.push(mapMenuItemKeyToFunction[itemKey](delegate));
+    } else {
+      console.log("Method/Function does not exist, skip: ", itemKey, methodName);
+    }
+  }
+
+  menuBarArray.push(getHelpMenuItems()); // Help-Menu at the end.
+  const menuBar = new MenuBar(menuBarArray);
+  return menuBar;
+}
+
+function getFontraMenuItems() {
+  return {
+    title: "Fontra",
+    bold: true,
+    getItems: () => {
+      const menuItems = [
+        "shortcuts",
+        "theme-settings",
+        "display-language",
+        "clipboard",
+        "editor-behavior",
+        "plugins-manager",
+        "server-info",
+      ];
+      return menuItems.map((panelID) => ({
+        title: translate(`application-settings.${panelID}.title`),
+        enabled: () => true,
+        callback: () => {
+          window.open(`/applicationsettings/applicationsettings.html#${panelID}-panel`);
+        },
+      }));
+    },
+  };
+}
+
+function getHelpMenuItems() {
+  return {
+    title: translate("menubar.help"),
+    getItems: () => {
+      return [
+        {
+          title: translate("menubar.help.homepage"),
+          enabled: () => true,
+          callback: () => {
+            window.open("https://fontra.xyz/");
+          },
+        },
+        {
+          title: translate("menubar.help.documentation"),
+          enabled: () => true,
+          callback: () => {
+            window.open("https://docs.fontra.xyz");
+          },
+        },
+        {
+          title: translate("menubar.help.changelog"),
+          enabled: () => true,
+          callback: () => {
+            window.open("https://fontra.xyz/changelog.html");
+          },
+        },
+        {
+          title: "GitHub",
+          enabled: () => true,
+          callback: () => {
+            window.open("https://github.com/googlefonts/fontra");
+          },
+        },
+      ];
+    },
+  };
+}
+
+function getFileMenuItems(delegate) {
```

Comment:
Oh, we had a misunderstanding here, that is shown in the function name. These functions return menu descriptors, not menu *items*. The delegate (viewController) should *not* be responsible for the whole menu, but only for the getItems() part. Hence the name.

With the Fontra and Help menu's it's differnt: it is useful to build the whole menu in a helper func like you do, but the function names are bad. `getFontraMenuItems()` doesn't return items, but a whole menu. So it should be called `getFontraMenu()`. Same with the Help menu.

Reply:
So the fallback functions (like `getFileMenuItems()`) should probably _also_ just do the "getItems()" part.

Reply:
So, I made the 'get' functions return only items, not descriptors. But `getFileMenuItems` does not work anymore, because `viewController.fontController.backendInfo` is `undefined` and I don't know why. It makes no sense to me. I do a little break now.

Reply:
Is fixed now.